app/mailers/report_mailer.rb

```diff
@@ -5,10 +5,29 @@ def budget_status
     to_email = ENV.fetch('BUDGET_REPORT_RECIPIENT')
     date = Time.now.utc.strftime("%a %m/%d/%y (%Z)")
 
+    build_attachments
+
     mail(
       to: to_email,
       subject: "C2: Daily Budget report for #{date}",
       from: self.sender_email
     )
   end
+
+  private
+
+  # rubocop:disable AbcSize, TrailingComma
+  def build_attachments
+    date = Time.now.utc.strftime('%Y-%m-%d')
+    { 'approved-ba60-week' => Ncr::Reporter.as_csv(Ncr::Reporter.ba60_proposals),
+      'approved-ba61-week' => Ncr::Reporter.as_csv(Ncr::Reporter.ba61_proposals),
+      'approved-ba80-week' => Ncr::Reporter.as_csv(Ncr::Reporter.ba80_proposals),
+      'pending-at-approving-official' => Ncr::Reporter.as_csv(Ncr::Reporter.proposals_pending_approving_official),
+      'pending-at-budget' => Ncr::Reporter.as_csv(Ncr::Reporter.proposals_pending_budget),
+      'pending-at-tier-one-approval' => Ncr::Reporter.as_csv(Ncr::Reporter.proposals_tier_one_pending),
+    }.each do |name, csv|
+      attachments[name + '-' + date + '.csv'] = csv
+    end
```

Comment:
This can be simplified by doing the CSV conversion in the loop:

```ruby
{ 'approved-ba60-week' => Ncr::Reporter.ba60_proposals,
  'approved-ba61-week' => Ncr::Reporter.ba61_proposals,
  'approved-ba80-week' => Ncr::Reporter.ba80_proposals,
  'pending-at-approving-official' => Ncr::Reporter.proposals_pending_approving_official,
  'pending-at-budget' => Ncr::Reporter.proposals_pending_budget,
  'pending-at-tier-one-approval' => Ncr::Reporter.proposals_tier_one_pending,
}.each do |name, proposals|
  csv = Ncr::Reporter.as_csv(proposals)
  attachments[name + '-' + date + '.csv'] = csv
end
```